src/helpers/glpTEST/getGlpSwapData.js

```diff
@@ -0,0 +1,318 @@
+    }
+    //
+
+    mimLeftToSwap = mimLeftToSwap.sub(awailableToSwap);
+  }
+
+  console.log("cookInfo", cookInfo);
+  console.log("result", tokenInfoArray);
+
+  for (let info of cookInfo) {
+    const oracleExchangeRate = pool.oracleExchangeRate;
+
+    const slippageBN = ethers.BigNumber.from(
+      parseFloat(slipage * 1e10).toFixed(0)
+    );
+
+    const userAddr = store.getters.getAccount;
+    const swapAmount = info.sellAmount;
+
+    const expectedCollateral = swapAmount
+      .mul(oracleExchangeRate)
+      .div(String(1e18));
+
+    const slippageAmount = expectedCollateral
+      .div(100)
+      .mul(slippageBN)
+      .div(1e10);
+    const minExpected = expectedCollateral.sub(slippageAmount);
+
+    const swapData = info.swapDataEncode;
+    const swapperAddres = pool.levSwapperContract.address;
+
+    const swapStaticTx = await pool.levSwapperContract.populateTransaction.swap(
+      userAddr,
+      minExpected,
+      swapAmount,
+      swapData
+    );
+
+    cookData = await actions.call(
+      cookData,
+      swapperAddres,
+      swapStaticTx.data,
+      false,
+      false,
+      2
+    );
+  }
+
+  console.log("cookData", cookData);
+
+  return cookData;
+};
+
+const getGlpLiqData = async (provider, pool, amount, chainId, slipage) => {
+  store.commit("updateRouteData", []);
+  store.commit("setPopupState", {
+    type: "test",
+    isShow: true,
+  });
+
+  const { borrowToken, collateralToken } = pool;
+  const gmxLensContract = await new ethers.Contract(
+    gmxLensAddress,
+    JSON.stringify(gmxLensAbi),
+    provider
+  );
+
+  const results = [];
+  const mGlpAmount = await pool.masterContractInstance.toAmount(
+    collateralToken.address,
+    amount,
+    false
+  );
+  const glpAmount = await collateralToken.contract.convertToAssets(mGlpAmount);
+  console.log("glpAmount", glpAmount.toString());
+  for (let token of await getTokens(provider)) {
+    const resp = await gmxLensContract.getTokenOutFromBurningGlp(
+      token,
+      glpAmount
+    );
+
+    const { buyAmount, sellAmount, data } = await swap0xRequest(
+      chainId,
+      borrowToken.address,
+      token,
+      slipage,
+      resp[0].toString()
+    );
+
+    // console.log("outAmount", resp[0].toString())
+    // console.log("buyAmount", buyAmount)
+    // console.log("feesPoints", resp[1].toString())
+
+    const getSwapDataEncode = ethers.utils.defaultAbiCoder.encode(
+      ["bytes", "address"],
+      [data, token]
+    );
+
+    results.push({
+      tokenName: getNameFromAddress(token),
+      token,
+      tokenOutAmount: resp[0].toString(),
+      buyAmount,
+      feesPoints: resp[1].toString(),
+      swapDataEncode: getSwapDataEncode,
+      sellAmount,
+    });
+
+    store.commit("updateRouteData", results);
+  }
+
+  const result = results.reduce(maxBuyAmount);
+
+  const notification = {
+    msg: `0x sell amount: ${result.sellAmount}; OutFromBurningGlp amount: ${result.tokenOutAmount}`,
+    type: "warning",
```

Comment:
why is the type of a warning? is this not more informational?

Reply:
it's just for the test originally, I'll remove it